pulp_2to3_migration/app/models/base.py

```diff
@@ -11,24 +17,113 @@ class MigrationPlan(Model):
         plan (models.JSONField): The migration plan in the JSON format
     """
     plan = JSONField()
+    _real_plan = None
+
+    @property
+    def plan_view(self):
+        """
+        Window to view the validated migration plan data through.
+        """
+        # Needs a MongoDB connection initialized beforehand to look for importers,
+        # verify existance of repositories and distributors, etc.
+        if not self._real_plan:
+            self._real_plan = _ValidatedMigrationPlan(self.plan)
+
+        return self._real_plan
+
+    def get_plugins(self):
+        """
+        Return a list of pulp2 plugins to migrate.
+        """
+        return self.plan_view.plugins_to_migrate
 
     def get_repositories(self):
         """
         Return a list of pulp2 repositories to migrate or empty list if all should be migrated.
         """
-        # TODO: get pulp2 repositories from the self.plan
-        return []
+        return self.plan_view.repositories_to_migrate
```

Comment:
@dralley , I thought we decided on IRC that to create repo versions we need a structure for repositories and that just a list of ids is not enough.
For importers/distributors, it's sufficient, however for repositories we need all the info to create a repo version, to create a distribution and so on. 
E.g. repositories can be a dict with repo name as a key.

Plan:
```
"repositories":
[
           {
             "name": "foo",
             "pulp2_repository_id":  "my_repo_id to use to find an importer to migrate"   # optional?
             "repository_versions": [
                 {
                   "pulp2_repository_id": "idA",
                   "distributor_ids":  ["distributor_id1", "distributor_id2"]  # optional?
                   "protection": ["distributor_id1"] # content protection migration is only needed for the specfied distributor(s)
                  },                
                  {
                    "pulp2_repository_id": "idB",
                    "distributor_ids":  ["distributor_id3"]
                  },
               ]
            }
]
```

get_repositories returns a dict with a repo_id as a key:
```
{ "foo": 
      {"importer_from_repo":  "my_repo_id to use to find an importer to migrate",
       "repo_versions": [  as they are in migration plan ]
      }
}
```

What do you think?